client/src/lib/firebase/auth.ts

```diff
@@ -35,14 +38,78 @@ export async function loginAnonymous() {
   return await signInAnonymously(auth);
 }
 
-export function deleteAccount() {
+export async function deleteAccount() {
   const user = auth.currentUser;
-
   if (user == null) {
     throw new Error("User is not defined.");
   }
 
-  deleteUser(user);
+  await deleteUser(user);
+}
+
+export function linkWithGoogle(user: User | null) {
+  if (user === undefined) {
+    throw new Error("Not signed in");
+  }
+
+  if (user != null) {
+    const google = new GoogleAuthProvider();
+    return linkWithPopup(user, google);
+  }
+}
+
+export function linkWithMicrosoft(user: User | null) {
+  if (user === undefined) {
+    throw new Error("Not signed in");
+  }
+
+  if (user != null) {
+    const microsoft = new OAuthProvider("microsoft.com");
+    return linkWithPopup(user, microsoft);
+  }
+}
+
+export function linkWithPassword(password: string) {
+  const user = auth.currentUser;
+  if (user === undefined) {
+    throw new Error("Not Signed In");
+  }
+
+  if (user != null) {
+    return updatePassword(user, password);
+  } else {
+    return user;
+  }
```

Comment:
if you are checking this in the if that throws an error you don't need to check it here